cnxarchive/views.py

```diff
@@ -981,3 +1008,45 @@ def robots(request):
     resp.expires = html_date(exp_time)
     resp.body = robots_dot_txt.to_string()
     return resp
+
+
+@view_config(route_name='recent', request_method='GET',
+             renderer='templates/recent.rss')
+def recent(request):
+    # setting the query variables
+    num_entries = request.GET.get('number', 10)
+    start_entry = request.GET.get('start', 0)
+    portal_type = request.GET.get('type', ['Collection', 'Module'])
+    if portal_type != ['Collection', 'Module']:
+        portal_type = [portal_type]
+    # search the database
+    settings = request.registry.settings
+    statement = """
+                SELECT name, revised, authors, abstract, uuid
+                FROM latest_modules
+                JOIN abstracts
+                ON latest_modules.abstractid = abstracts.abstractid
+                WHERE portal_type = ANY (%s)
+                ORDER BY revised DESC
+                LIMIT (%s) OFFSET (%s);
+                """
+    with psycopg2.connect(settings[config.CONNECTION_STRING]) as db_connection:
+            with db_connection.cursor() as cursor:
+                cursor.execute(statement,
+                               vars=(portal_type, num_entries, start_entry))
+                search_results = cursor.fetchall()
+    # alter the results to be in the correct form
+    latest_modules = []
+    titles = ["name", "revised", "authors", "abstract", "uuid"]
+    for i in range(len(search_results)):
```

Comment:
@reedstrm Is this change necessary? Because when I change it to db_connection.cursor(cursor_factory=psycopg2.extras.DictCursor) and try to cursor.fetchall() it is giving me a list and only returns the class 'psycopg2.extras.DictRow' type when I use fetchone() which then seems like more work to fetch it one at a time